src/lib/provable/int.ts

```diff
@@ -1842,6 +1842,15 @@ class UInt8 extends Struct({
     return new UInt8(x);
   }
 
+  static fromValue(
```

Comment:
In the case of `fromField(ff)`, isn't this similar to `new UInt8(ff.value)` or `UInt8.from(ff)`, which are not recommended to use instead of `UInt8.Unsafe.fromField(ff)` because they don't clearly indicate the potential unsafety of wrapping a Field into a UInt8, making the code less readable.

Reply:
yeah absolutely, and this method isn't really intended for developer to use (that's what `.from()` was designed for).

`fromValue()` is part of the `Provable<T>` API and what we use internally to convert JS values into the type -- it's needed here for that reason.

Note that this PR doesn't add `UInt8.fromValue()` -- it was already there, we just override it to also handle `number` inputs (which is not unsafe), because `number` is the most natural type that you'd want to convert a UInt8 from

Reply:
there are two things to improve IMO, both of which aren't related to this PR though:
1. internal `Provable` methods should be on a separate namespace `.provable` instead of on the class itself, to mark then more clearly as not-developer facing https://github.com/o1-labs/o1js/issues/1248
2. `UInt8` shouldn't be a `Struct`, because while `Struct` is great for easily _composing_ types, it's not great for defining the base primitives. In this case, the problem is that the `fromValue()` type of `UInt8` is inferred from the `Struct({ value: Field })` definition to include `{ value: Field }`, which really should be `UInt8` (but that doesn't work because the Struct definition doesn't know anything about `UInt8`)

Reply:
>  isn't this similar to new UInt8(ff.value) or UInt8.from(ff), which are not recommended to use

side note: `UInt8.from(field)` is perfectly fine, because it adds a range-check

Reply:
Thanks for the insight! 

> UInt8 shouldn't be a Struct, because while Struct is great for easily composing types, it's not great for defining the base primitives.

I recall encountering an issue a while ago where I couldn't use `Provable.witness` for a `UInt32` with `proofsEnabled=true`. Is this related to the same issue, or is it unrelated? I’ll need to revisit and check this again!

Reply:
`UInt32` is not implemented with `Struct` (but with the older `CircuitValue`) so that might be a different issue!

in the case of `UInt8`, one thing that bites me now and then is that `UInt8 satisfies Provable<UInt8>` is not true (and similar for other Structs), because e.g. the return type of `UInt8.fromFields()` is `{ value: Field }` and not `UInt8`

![image](https://github.com/user-attachments/assets/b92015f0-a8e0-44e9-89d1-31310e5d0f0c)

Reply:
While if you look at newer types like `Bytes`, you'll see that because of the way we made `.provable` aware of the class itself, this works:

```ts
Bytes(32) satisfies ProvableType<Bytes>;
```